File: E-Booking/E-Booking.Models/TransportRequirementModel.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Web;
using System.ComponentModel.DataAnnotations;
using ExpressiveAnnotations.Attributes;

namespace EBooking.Models
{
    public class TransportRequirementModel
    {
        
        public int Id { get; set; }

        [Required(ErrorMessage = "Transport Request Reason is required.")]
        public int TransportRequestReasonId { get; set; }

        [Required(ErrorMessage = "Transport Selection is required.")]
        public int TransportSelectionId { get; set; }
        
        public bool IsInfectious { get; set; }

        [RequiredIf("IsInfectious == true", ErrorMessage = "Infectious is required.")]
        public int InfectiousId { get; set; }

        [RequiredIf("InfectiousId == 4 && IsInfectious == true", ErrorMessage = "Additional Patient Info is required.")]
        //[StringLength(200, ErrorMessage = "Additional Patient Info cannot be longer than 200 characters and Less than 10 characters.", MinimumLength = 10)]
        public string AdditionalPatientInfo { get; set; }

        public bool IsTravellingWithOwnOxygen { get; set; }
        
        public bool IsEscortTravelling { get; set; }

        [RequiredIf("IsEscortTravelling == true", ErrorMessage = "Escort Type is required.")]
        public int EscortTypeId { get; set; }

        [RequiredIf("IsEscortTravelling == true", ErrorMessage = "Escort Number is required.")]
        public int EscortNumberId { get; set; }

        public bool IsBariatric { get; set; }

        public bool IsFullLegPlasterPOP { get; set; }

        public bool IsElectricWheelchair { get; set; }

        public bool IsWheelchairAndLegExtension { get; set; }

        public bool IsWaterlow { get; set; }

        public bool IsDNACPR { get; set; }

        public bool IsDiabetic { get; set; }

        public bool IsNuclearMedicineRadioActiveRisk { get; set; }

        public bool IsNoneOfAbove { get; set; }

        public bool Is24HourAmendment { get; set; }

    }
}

File: E-Booking/E-Booking/Scripts/KnockoutModels/TransportRequirementModel.js
function TransportRequirementModel()
{
    var self = this;
    //debugger;
    self.Id = ko.observable(0);
    self.TransportRequestReasonId = ko.observable().extend({ required: true });
    self.TransportSelectionId = ko.observable().extend({ required: true });
    
    //Additional Patient Informations
    self.IsTravellingWithOwnOxygen = ko.observable(false);
    
    self.IsInfectious = ko.observable(false); 
    self.InfectiousId = ko.observable().extend({ required: { onlyIf: function () { return self.IsInfectious(); } } });
    self.IsInfectious.subscribe(function (value) { if (value == false) { self.InfectiousId(null); } });
    self.AdditionalPatientInfo = ko.observable('').extend({ required: { onlyIf: function () { return (self.InfectiousId() == 4 && self.IsInfectious() == true ? true : false); } }, minLength: 10 });
    self.IsEscortTravelling = ko.observable(false);
    self.EscortTypeId = ko.observable().extend({ required: { onlyIf: function () { return self.IsEscortTravelling(); } } });
    self.EscortNumberId = ko.observable().extend({ required: { onlyIf: function () { return self.IsEscortTravelling(); } } });
    self.IsEscortTravelling.subscribe(function (value) { if (value == false) { self.EscortTypeId(null); self.EscortNumberId(null); } });
    self.IsBariatric = ko.observable(false);
    self.IsFullLegPlasterPOP = ko.observable(false);
    self.IsElectricWheelchair = ko.observable(false);
    self.IsWheelchairAndLegExtension = ko.observable(false);
    self.IsWaterlow = ko.observable(false);
    self.IsDNACPR = ko.observable(false);
    self.IsDiabetic = ko.observable(false);
    self.IsNuclearMedicineRadioActiveRisk = ko.observable(false);
    self.IsNoneOfAbove = ko.observable(false);

    self.IsInfectious.extend({ checked: { onlyIf: function () { return CheckAtleastOneOptionSelected(self); }, message: "Please select at least one option to proceed." } });
    self.IsTravellingWithOwnOxygen.extend({ checked: { onlyIf: function () { return CheckAtleastOneOptionSelected(self); }, message: "Please select at least one option to proceed." } });
    self.IsEscortTravelling.extend({ checked: { onlyIf: function () { return CheckAtleastOneOptionSelected(self); }, message: "Please select at least one option to proceed." } });
    self.IsBariatric.extend({ checked: { onlyIf: function () { return CheckAtleastOneOptionSelected(self); }, message: "Please select at least one option to proceed." } });
    self.IsFullLegPlasterPOP.extend({ checked: { onlyIf: function () { return CheckAtleastOneOptionSelected(self); }, message: "Please select at least one option to proceed." } });
    self.IsElectricWheelchair.extend({ checked: { onlyIf: function () { return CheckAtleastOneOptionSelected(self); }, message: "Please select at least one option to proceed." } });
    self.IsWheelchairAndLegExtension.extend({ checked: { onlyIf: function () { return CheckAtleastOneOptionSelected(self); }, message: "Please select at least one option to proceed." } });
    self.IsWaterlow.extend({ checked: { onlyIf: function () { return CheckAtleastOneOptionSelected(self); }, message: "Please select at least one option to proceed." } });
    self.IsDNACPR.extend({ checked: { onlyIf: function () { return CheckAtleastOneOptionSelected(self); }, message: "Please select at least one option to proceed." } });
    self.IsDiabetic.extend({ checked: { onlyIf: function () { return CheckAtleastOneOptionSelected(self); }, message: "Please select at least one option to proceed." } });
    self.IsNuclearMedicineRadioActiveRisk.extend({ checked: { onlyIf: function () { return CheckAtleastOneOptionSelected(self); }, message: "Please select at least one option to proceed." } });
    self.IsNoneOfAbove.extend({ checked: { onlyIf: function () { return CheckAtleastOneOptionSelected(self); }, message: "Please select at least one option to proceed." } });
    self.Is24HourAmendment = ko.observable(false);
    self.errors = ko.validation.group(self);
}

function CheckAtleastOneOptionSelected(self) {
    return !(self.IsInfectious() || self.IsEscortTravelling() || self.IsTravellingWithOwnOxygen() || self.IsBariatric() || self.IsFullLegPlasterPOP() || self.IsElectricWheelchair() || self.IsWheelchairAndLegExtension() || self.IsWaterlow() || self.IsDNACPR() || self.IsDiabetic() || self.IsNuclearMedicineRadioActiveRisk() || self.IsNoneOfAbove());
}
